TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -506,7 +506,25 @@ public static DataDefinition CheckVariable(Node node, StorageArea storageArea, b
                         node.GetEnclosingProgramOrFunctionNode().SetFlag(Node.Flag.UseGlobalStorage, true);
                     }
                 }
-                
+
+                if (!isReadStorageArea && node.SymbolTable.CurrentScope == SymbolTable.Scope.Function)
+                {
+                    ParameterDescription paramDesc;
+                    if (dataDefinitionPath?.CurrentDataDefinition.TypeDefinition != null)
```

Comment:
The if/else can be simplified :
paramDesc  = (dataDefinitionPath.CurrentDataDefinition ?? dataDefinitionFound) as ParameterDescription;

Reply:
done
https://github.com/TypeCobolTeam/TypeCobol/tree/55e817f33a3d27d5c35869ccddd44cd00f20b192